pkg/plugins/resources/cargopackage/source.go

```diff
@@ -0,0 +1,44 @@
+package cargopackage
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/result"
+)
+
+// Source returns the latest npm package version
+func (cp CargoPackage) Source(workingDir string) (string, error) {
+	if len(cp.spec.IndexDir) == 0 {
+		pwd, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+		if workingDir == pwd {
+			// No IndexDir provided nor scm resource
+			// We should use the default `crates.io` repository
+			indexDir, err := loadDefaultCrateIndex()
+			if err != nil {
+				return "", err
+			}
+			cp.spec.IndexDir = indexDir
+		} else {
+			cp.spec.IndexDir = workingDir
+		}
+	}
+
```

Comment:
This is in order to detect wether or not we are getting the source in the context of `scm` or not. Is there a better way to do it?

Reply:
I didn't test it but something like 

```
package cargopackage

import (
	"github.com/updatecli/updatecli/pkg/plugins/scms/git"
)

func cloneCrateGitRepository() (string, error) {

	s, err := git.New(
		git.Spec{
			URL: "https://github.com/rust-lang/crates.io-index.git",
		},
	)

	if err != nil {
		return "", err
	}

	return s.Clone()
}
```

Reply:
I just did some testing and while it works, it just super slow like a minute
I didn't try your approach

Reply:
I agree, the crates.io repo is huge, so when we pull it it takes time. Hence why I wanted to let the user use a scm resource to do it.

if `go-git` would allow sparse checkout we could only download the crate file, but the issue is pending there

Reply:
What surprise me is even a repository that has already been 

First time repository download 
- `./bin/updatecli diff --config e2e/updatecli.d/success.d/crate.yaml --debug  289.71s user 22.11s system 128% cpu 4:03.55 total`

Second time once the repository is already cloned
- `./bin/updatecli diff --config e2e/updatecli.d/success.d/crate.yaml --debug  173.48s user 20.45s system 145% cpu 2:13.04 total
`

I am bit surprised that it doesn't go much faster once already downloaded.  Usually repositories that I work on are relatively small so I am never  affected by such slowness maybe I could do some optimization on Updatecli side.

I am testing a very simple config file

```
sources:
  test1:
    kind: cargopackage
    spec:
      package: syn 
  test2:
    kind: cargopackage
    spec:
      package: rand
```

Reply:
I am wondering if we can also leverage `$CARGO_HOME/registry` 
Something like:

1. If scm then use `scm.getDirectory()` if scm is specified
2. If indexDir use `index.Dir` if it's specified
3. If `$CARGO_HOME/registry` if it exists
4. Clone `$CARGO_HOME/registry` in `/tmp/updatecli`

Reply:
I don't see how to improve the clone operation but the crates.io repository is constantly updated